### PR TITLE
Tracking

### PR DESCRIPTION
**Pull request by maraeckart**

### CODE REVIEW

File: src/train_binary.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        transforms.ToTensor(),
+        transforms.Normalize([0.485,0.456,0.406],[0.229,0.224,0.225]),
+    ])
+
+    tf_test = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize([0.485,0.456,0.406],[0.229,0.224,0.225]),
+    ])
+
+    train_ds = datasets.ImageFolder(Path(args.data_root)/"train", transform=tf_train)
```
Comment (codewithlaurin) [inline src/train_binary.py]: Wieso wird hier nicht der Binary folder verwendet?

File: src/train_multiclass.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            "train/loss": train_loss,
+            "train/acc": train_acc,
+            "train/f1": train_f1,
+        })
+
+    model.eval()
+    total, correct = 0, 0
+    test_preds, test_targets = [], []
+    
+    idx_to_name = {i: name for i, name in enumerate(train_ds.classes)}
```
Comment (codewithlaurin) [inline src/train_multiclass.py:95]: Du solltest du das erstellen der Tabelle in eine eigene Funktion auslagern.

File: src/train_multiclass.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+from sklearn.metrics import f1_score
+import wandb 
+import torch.nn.functional as F
+from torchvision.transforms.functional import to_pil_image
+
+def get_device():
+    if torch.cuda.is_available(): return torch.device("cuda")
+    if hasattr(torch.backends, "mps") and torch.backends.mps.is_available(): return torch.device("mps")
+    return torch.device("cpu")
+
```
Comment (codewithlaurin) [inline src/train_multiclass.py:15]: Add deterministic seeding

torch.manual_seed(42)
np.random.seed(42)
random.seed(42)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False

File: src/train_multiclass.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    model = models.resnet18(weights=models.ResNet18_Weights.IMAGENET1K_V1)
+    model.fc = nn.Linear(model.fc.in_features, len(train_ds.classes))
+    model.to(device)
+
+    criterion = nn.CrossEntropyLoss()
+    # optim = torch.optim.Adam(model.parameters(), lr=args.lr)
+    optim = torch.optim.SGD(model.parameters(), lr=args.lr, momentum=0.9)
+    wandb.watch(model, criterion, log="all", log_freq=100)
+
+    for epoch in range(1, args.epochs+1):
```
Comment (codewithlaurin) [inline src/train_multiclass.py:59]: Man könnte noch in jeder epoche den loss und die accuracy berechnen und auf wandb speichern

**Review by codewithlaurin — CHANGES_REQUESTED** [review]
